targets/psa_switch/externs/psa_internet_checksum.h

```diff
@@ -0,0 +1,60 @@
+/* Copyright 2021 SYRMIA LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/*
+ * Dusan Krdzic (dusan.krdzic@syrmia.com)
+ *
+ */
+
+
+#ifndef PSA_SWITCH_PSA_INTERNETCHECKSUM_H_
+#define PSA_SWITCH_PSA_INTERNETCHECKSUM_H_
+
+#include <bm/bm_sim/extern.h>
+
+namespace bm {
+
+namespace psa {
+
+class PSA_InternetChecksum : public bm::ExternType {
```

Comment:
there should be unit tests for this class

Reply:
I tested it with `.stf` and `p4c` tests. But if it is necessary I will add it.
`p4c test` : https://github.com/p4lang/p4c/pull/2850/files#diff-99faa3a1936103edeab4cd96fe10b8346c60378ff8e0807a75f737d79b54cb3a
`stf` : https://github.com/p4lang/p4c/pull/2850/files#diff-bb85f831658a92ca319ef52362eb1254e44e452638e99911e6cbeca268d16725

Reply:
there should be some level of basic testing (as google unit tests) to gate future changes to this extern implementation. This is the first extern implementation for PSA that is not totally trivial.

Reply:
ping on this